docs/backup_and_restore.rst

```diff
@@ -101,89 +88,257 @@ When you are ready to begin the backup, run
+configuration will only include v3 onion services regardless of the backup state.
+
+The restore process differs based on the onion services that were configured on
+the old instance and preserved in the backup:
+
+  - :ref:`Migrating using a v2+v3 or v3-only backup <migrate_v3>`
+  - :ref:`Migrating using a v2-only backup <migrate_v2>`
+
+.. _migrate_v3:
+
+Migrating Using a V2+V3 or V3-Only Backup
+'''''''''''''''''''''''''''''''''''''''''
+
+.. include:: includes/v3_backup_steps_common.rst
+
+.. _migrate_v2:
+
+Migrating Using a V2-Only Backup
+''''''''''''''''''''''''''''''''
+
+V2 onion services are no longer supported for new SecureDrop installs, so
+*Source* and *Journalist Interface* addresses will change when you perform a
+migration using a v2-only backup. However, it is possible to migrate submissions,
+source accounts, and journalist accounts. To do so, follow the steps below:
+
+.. note:: The instructions below assume that you are using the same *Admin Workstation*
+  that was used to manage your old instance. If you are using a new *Admin
+  Workstation* you will need to copy the directory ``~amnesia/Persistent/securedrop``
+  from the old workstation to the new workstation (using a *Transfer Device*)
+  before proceeding.
+
+#. If you have not already done so, :ref:`back up the existing installation <backing_up>`. The instructions below assume that the backup has been created and renamed ``sd-backup-old.tar.gz``.
 
-Make sure to replace ``sd-backup-2017-07-22--01-06-25.tar.gz`` with the filename
-for your backup archive. The backup archives are located in
-``install_files/ansible-base``.
+#. Move the existing *Admin Workstation* SecureDrop code out of the way, by
+   opening a Terminal via **Applications > System Tools > Terminal** and
+   running the command:
 
-Once the restore is done, the *Application Server* will use the original
-*Source Interface* and *Journalist Interface* Onion URLs. You will need
-to update the corresponding files on the *Admin Workstation*:
+   .. code:: sh
 
-* ``app-source-ths``
-* ``app-journalist-aths``
-* ``app-ssh-aths``
+      mv ~/Persistent/securedrop ~/Persistent/sd.bak
 
-Once SSH access to the servers has been established (or if using SSH over
-local network), Onion URLs for the *Source Interface* and *Journalist Interfaces*
-can be fetched using the installer:
+#. Move the existing *Admin Workstation* SSH configuration out of the way via
+   the Terminal, using the commands:
+
+   .. code:: sh
+
+       mv ~/.ssh/config ~/.ssh/config.bak
+       mv ~/.ssh/known_hosts ~/.ssh/known_hosts.bak
+
+#. Reinstall SecureDrop  on the *Admin Workstation* using the following Terminal
+   commands:
+
+   .. code:: sh
+
+      cd ~/Persistent
+      git clone https://github.com/freedomofpress/securedrop
+
+#. Verify that the current release tag was signed with the release signing key:
+
+   .. code:: sh
+
+      cd ~/Persistent/securedrop/
+      git fetch --tags
+      git tag -v 1.7.1
```

Comment:
This version string is not encapsulated by your changes to `update_version.sh` (it will remain at 1.7.1 after bumping the version).

Reply:
I noticed this (the changes you are remarking on are from @zenmonkeykstop 's patch). I'm currently updating those. However, rst is not playing nice with the `|version|` placeholder when it is inlined in a numbered list and surrounded by the `.. warning` callout; it renders half of the `version`s correctly and half of them incorrectly. I'll see if I can resolve this, unless you are a Sphinx genius in which case please advise :)

Reply:
This is a big part of the reason why `update_version.sh` exists -- unfortunately we can't straightforwardly use `|version|` in inline snippets, so in those cases the script goes in and uses `sed` to perform a substitution. If you just add a similar substitution to the one already added for `backup_and_restore.rst` to the script, we should be golden.